--- a/includes/ResourceLoaderWikiMarkdownVisualEditorModule.php
+++ b/includes/ResourceLoaderWikiMarkdownVisualEditorModule.php
@@ -1,11 +1,11 @@
 <?php

-class ResourceLoaderWikiMarkdownVisualEditorModule extends ResourceLoaderFileModule {
+class ResourceLoaderWikiMarkdownVisualEditorModule extends MediaWiki\ResourceLoader\FileModule {

 	protected $targets = [ 'desktop', 'mobile' ];

 	/**
-	 * @param ResourceLoaderContext $context
+	 * @param MediaWiki\ResourceLoader\Context $context
 	 * @return string JavaScript code
 	 */
-	public function getScript( ResourceLoaderContext $context ) {
+	public function getScript( MediaWiki\ResourceLoader\Context $context ) {
 		$scripts = parent::getScript( $context );

 		return $scripts;